src/Microsoft.OData.Core/UriParser/Resolver/ODataUriResolver.cs

```diff
@@ -124,15 +134,23 @@ public virtual IEdmProperty ResolveProperty(IEdmStructuredType type, string prop
             }
 
             var result = type.Properties()
-            .Where(_ => string.Equals(propertyName, _.Name, StringComparison.OrdinalIgnoreCase))
-            .ToList();
+            .Where(_ => string.Equals(propertyName, _.Name, StringComparison.OrdinalIgnoreCase));
+
+            IEdmProperty resolvedProperty = default(IEdmProperty);
 
-            if (result.Count > 1)
+            foreach(var candidate in result)
             {
-                throw new ODataException(Strings.UriParserMetadata_MultipleMatchingPropertiesFound(propertyName, type.FullTypeName()));
+                if (resolvedProperty == default(IEdmProperty))
+                {
+                    resolvedProperty = candidate;
+                }

```

Comment:
1. the codes here look unreadable. 

2. we can't use "null" to replace default(IEdmProperty)

3) SingelOrDefault will evaluate the Linq, and it's readable and maintainable, so, what's the percentage to replace it like the codes here?